### DISCUSSION TITLE
Highlight active file

### DISCUSSION THREAD

**Discussion by JoseConseco** [nvim-tree/nvim-tree.lua#2001] [Q&A]
Nvim tree can highlight opened files, and unfold required folders, but I would like it to highlight active file - eg. make active file bground brighter. Is there option for it? highlight_opened_files - seems to show all opened files, not the current one...

**Comment by alex-courtis** [comment]
Does `:help nvim-tree.update_focused_file` meet your needs?

**Reply by JoseConseco** [reply]
Not quite - as mentioned I wish to have way to differentiate the  ACTIVE buffer only  (with new custom highlight 'NvimTreeActiveFile' ).   Update focused files - shows me all opened files, but I still cannot easily spot active buffer: 
![image](https://user-images.githubusercontent.com/13521338/219866416-56f513f8-ccc2-41d2-9281-a4b337529ead.png)
I often have 10 on more files opened at same time,  so having them all highlighted does not help that much.

**Reply by alex-courtis** [reply]
This looks like a configuration issue. The cursorline will show the active buffer.

Do you have `&cursorline` set? Do you have `view.cursorline` set?

Here's a screenshot with all defaults and `renderer.highlight_opened_files = "all"`
![20230219_112019](https://user-images.githubusercontent.com/1505378/219905369-e348b039-97a6-436f-aa47-83d91453ed42.png)

`.luacheckrc` and `CONTRIBUTING.md` are open, the latter is the active buffer.

**Reply by xzbdmw** [reply]
Hello, if I set a fg color of NvimtreeCursorLine, file icon's color will be overwritten, are there any workaround for that?

**Reply by alex-courtis** [reply]
Yup, nvim will do that - clobber everything including foreground. You'll get the same result when setting a fg colour for `CursorLine`.

It's recommended not to make too many changes to cursor line to avoid overwriting things like syntax highlighting.

**Comment by JoseConseco** [comment]
Well  it seems the `NvimTreeCursorLine` highlight is only shown/rendered when my cursor is in NvimTree  buffer. As soon as I move cursor to  edit file, the highlight is gone. I have view.cursorline set,  and highlight_opened_files is 'All'.  Below u can see the cursor line in NvimTree is hidden as soon as I move cursor from NvimTree: 
[highlight_nvimtree.webm](https://user-images.githubusercontent.com/13521338/219948133-d39934be-f2cf-4e98-b08e-a3a02235ad33.webm)
Config:

```lua
require("nvim-tree").setup {
    reload_on_bufenter = true,
    renderer = { highlight_opened_files = "all" },
    sync_root_with_cwd = true,
    respect_buf_cwd = true,
    update_focused_file = {
        enable = true,
        update_root = true,
        ignore_list = {},
    },
    filters = { dotfiles = true },
    view = {
        cursorline = true,
        preserve_window_proportions = true,
        mappings = {
            custom_only = false,
            list = {
                { key = "l", action = "edit",           action_cb = edit_or_open },
                { key = "L", action = "vsplit_preview", action_cb = vsplit_preview },
                { key = "h", action = "close_node" },
                { key = "H", action = "collapse_all",   action_cb = collapse_all },
                -- { key = "H", action = "call_hydra", action_cb = function() Hydra.spawn('nvim_tree_hydra') end },
            },
        },
    },
    actions = {
        open_file = {
            quit_on_open = false,
        },
    },
}

```

**Reply by gegoune** [reply]
Have you, perhaps, got functionality such that only current window has `cursorline` enabled? This is something that I have seen in quite a few configurations. I do that as well and had to add extra condition to leave nvim-tree out of it.

```lua
vim.api.nvim_create_autocmd({ 'InsertLeave', 'WinEnter' }, {
  group = wins,
  callback = function()
    local ok, cl = pcall(vim.api.nvim_win_get_var, 0, 'auto-cursorline')
    if ok and cl then
      vim.wo.cursorline = true
      vim.api.nvim_win_del_var(0, 'auto-cursorline')
    end
  end,
})

vim.api.nvim_create_autocmd({ 'InsertEnter', 'WinLeave' }, {
  group = wins,
  callback = function(event)
    local cl = vim.wo.cursorline
    if cl and not event.file:match 'NvimTree' then
      vim.api.nvim_win_set_var(0, 'auto-cursorline', cl)
      vim.wo.cursorline = false
    end
  end,
})
```

**Reply by JoseConseco** [reply]
Thx, that did the trick. I have forgotten that that I had code for hiding cursor line.  IMO it would be 'better' if active buffer highlight was not dependent on cursorline, but I could use custom highlight - dependent only on nvim-tree config  (remove its dependency on cursorline  nvim config parameter). @alex-courtis  what do you think?

**Reply by gegoune** [reply]
I think it makes sense as moving around in tree will move cursor line. Do you mind opening an issue for that?

**Reply by JoseConseco** [reply]
Ok here it is - https://github.com/nvim-tree/nvim-tree.lua/issues/2013
